db/version_edit.cc

```diff
@@ -302,6 +378,22 @@ bool VersionEdit::EncodeTo(std::string* dst) const {
     PutVarint32(dst, kInAtomicGroup);
     PutVarint32(dst, remaining_entries_);
   }
+
+  for (const AddedWal& wal : new_wals_) {
+    PutVarint32(dst, kNewWal);
+    wal.EncodeTo(dst);
+  }
+
+  for (WalNumber log_number : archived_wals_) {
```

Comment:
Assuming we want to track these events, we could consider introducing a dedicated class `ArchivedWal` that would encapsulate the related logic.